### DISCUSSION TITLE
Set-PnPDefaultColumnValues Index was outside the bounds of the array

### DISCUSSION THREAD

**Discussion by MichaelGaardbo** [pnp/powershell#2149] [Q&A]
Hi,

I'm trying to set a default value on a managed metadata column on a subfolder however I get:
**Set-PnPDefaultColumnValues : Index was outside the bounds of the array.**

I've tried different approaches:
$DocumentType = "Group|TermSet|TermValue" or $DocumentType = "wssid;#label|guid" or $DocumentType = "Guid of Term"
$FolderPath = "/Folder1/Folder2" or "Folder1/Folder2":
**Set-PnPDefaultColumnValues -List "ListName" -Field "FieldInternalName" -Folder $FolderPath -Value $DocumentType -Connection SPsiteConnection**

I've tried on a Single line of text column as well and same result:
**Set-PnPDefaultColumnValues : Index was outside the bounds of the array.**
On a side note: I played around with the -list parameter when I was testing and the strange thing is that if I put in a wrong value (non existing list) for -list it does NOT throw any exception.

Using: PnP.PowerShell 1.11.0

**Comment by denicomp** [comment]
Have you tried not using the "FieldInternalName" and use the Display Name of the field?

I am able to get Set-PnPDefaultColumnValues to work.  See the example code I put in the issue I opened yesterday:

https://github.com/pnp/powershell/issues/2143

My issue with Set-PnPDefaultColumnValues using managed metadata is that eventually, if used enough times, it will fail with the "Index was outside the bounds of the array". And it corrupts the default column values in the library, and the only way to fix it is to delete and rebuild the library.

In the example code I posted in the issue above, it happens after 3 calls.

This occurs when I use a Custom Property on a term and use it on folders more than 2 levels deep.

But it will eventually occur even without Custom Properties and folders 1 level deep, it just takes many more calls to Set-PnPDefaultColumnValues. But eventually it happens. There is no reason I can pinpoint,

But once it occurs (after successful calls), you can never use it again on that library - it will fail every time.

I do not see this behavior with Text fields however (so far). Only managed metadata fields.

**Reply by MichaelGaardbo** [reply]
Thank you the the reply, Denicomp.
This is exactly what I am experiencing it was working when testing stuff yesterday and today it's broke. I realized the code worked fine when run on another site...
The Clear-PnPDefaultColumnValues function does not work either - I had hoped to be able to clear the values and then be able to fill the columns with the script afterwards... No such luck.

**Reply by dkatruinsight** [reply]
This issue seem to be occurring even after deleting and recreating the library and trying to fire off Set-PnPDefaultColumValues command on the managed metadata column that was causing the problem in the first place. The workaround for me was to delete the library, recreate it, and manually set the default column value.

**Reply by adam13hylo** [reply]
Seeing the same issue here, even when trying to set the column default value on a single line of text field. I am currently using the following:

`Set-PnPDefaultColumnValues -List "Documents" -Field "projectName" -Folder "Folder 1" -Value "Project 1" -ErrorAction Stop`

**Reply by HiltonGiesenow** [reply]
Also seeing this, if anyone has a solution?

**Reply by TechsUK** [reply]
I am seeing this issue - I have tried on my document library with the managed metadata field and regular text fields now. I can set them through the UI, but no with Set-PnPDefaultColumnValues

**Comment by ssamnadda** [comment]
> Also seeing this, if anyone has a solution?

Was receiving the message "Index was outside the bounds of the array".  Upgrading to PnP.PowerShell 2.4.0 solved the issue for me.
